Smoketest failure seems unrelated:
src-react/components/WorkunitsDashboard.tsx(3,22): error TS2307: Cannot find module '@fluentui/react-cards' or its corresponding type declarations.
src-react/components/WorkunitsDashboard.tsx(12,10): error TS2305: Module '"../util/history"' has no exported member 'pushParamExact'.
make[2]: *** [esp/src/CMakeFiles/eclwatch.dir/build.make:331: esp/src/build/dist/stub.eclwatch.js] Error 1
make[1]: *** [CMakeFiles/Makefile2:12499: esp/src/CMakeFiles/eclwatch.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

--- a/dockerfiles/startall.sh
+++ b/dockerfiles/startall.sh
@@ -25,6 +25,7 @@ scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 restArgs=()
 
 CMD="install"
+DEP_UPDATE_ARG="--dependency-update"
 while [ "$#" -gt 0 ]; do
   arg=$1
   if [[ ${arg:0:1} == '-' ]]; then
@@ -40,11 +41,15 @@ while [ "$#" -gt 0 ]; do
       p) shift
          PERSIST=$1
          ;;
+      x) shift;
+          DEP_UPDATE_ARG=""
+         ;;
       h) echo "Usage: startall.sh [options]"
          echo "    -d <docker-repo>   Docker repository to fetch images from"
          echo "    -l                 Build image label to use"
-         echo "    -u                 Use "upgrade" rather than "install"
-         echo "    -p <location>      Use local persistent data
+         echo "    -u                 Use "upgrade" rather than "install""
+         echo "    -x                 Do not issue dependency-update"
+         echo "    -p <location>      Use local persistent data"
          exit
          ;;
       *) restArgs+=(${arg})
@@ -64,9 +69,9 @@ if [[ -n ${PERSIST} ]] ; then
   mkdir -p ${PERSIST}/dali
   mkdir -p ${PERSIST}/data
   helm ${CMD} localfile $scriptdir/../helm/examples/local/hpcc-localfile --set common.hostpath=${PERSIST} | grep -A100 storage > localstorage.yaml && \
-  helm ${CMD} mycluster $scriptdir/../helm/hpcc/ --set global.image.root="${DOCKER_REPO}" --set global.image.version=$LABEL --set global.privileged=true -f localstorage.yaml ${restArgs[@]}
+  helm ${CMD} mycluster $scriptdir/../helm/hpcc/ --set global.image.root="${DOCKER_REPO}" --set global.image.version=$LABEL --set global.privileged=true -f localstorage.yaml $DEP_UPDATE_ARG ${restArgs[@]}
 else
-  helm ${CMD} mycluster $scriptdir/../helm/hpcc/ --set global.image.root="${DOCKER_REPO}" --set global.image.version=$LABEL --set global.privileged=true ${restArgs[@]}
+  helm ${CMD} mycluster $scriptdir/../helm/hpcc/ --set global.image.root="${DOCKER_REPO}" --set global.image.version=$LABEL --set global.privileged=true $DEP_UPDATE_ARG ${restArgs[@]}
 fi
 
 sleep 1

--- a/helm/hpcc/Chart.yaml
+++ b/helm/hpcc/Chart.yaml
@@ -10,4 +10,23 @@ version: 0.1.0
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application.
+
 appVersion: 0.1.0
+
+# Requires 'helm dependency update' command
+dependencies:
+# An Elastic Stack environment is managed by default
+# To disable: --set elasticsearch.enabled=false --set kibana.enabled=false
+# and speficy alternate log processor
+- name: filebeat # Managed stderr log-forwarder
+  version: 7.9.3
+  repository: https://helm.elastic.co
+  condition: elasticsearch.enabled
+- name: elasticsearch # Managed logging processor engine
+  version: 7.9.3
+  repository: https://helm.elastic.co
+  condition: elasticsearch.enabled
+- name: kibana # Optional managed logging processor front-end
+  version: 7.9.3
+  repository: https://helm.elastic.co
+  condition: kibana.enabled #should only enable if elasticsearch.enabled=true

--- a/helm/hpcc/README.md
+++ b/helm/hpcc/README.md
@@ -1,0 +1,18 @@
+## HPCC Systems HELM Chart
+
+### Dependencies
+- HPCC Systems is designed to deploy a local, minimal Elastic Stack instance for log processing. This dependency requires a local copy of the Elastic Stack Helm charts for ElasticSearch, Filebeats and optionally Kibana. Helm provides a convenient command to pull appropriate dependencies locally:
+> helm dependency update
+
+- Note, the above command should be issued from the hpcc chart location.
+
+- Otherwise, provide the "--dependency-update" argument in the helm install command
+> For example:
+> helm install myhpcccluster . --set global.image.version --dependency-update
+
+- The managed Elastic Stack instance can be disabled by setting the following values to false:
+>--set elasticsearch.enabled=false
+and
+>--set kibana.enabled=false
+
+- Please note, even when above dependency flags are disabled, the helm install process enforces all declared dependency charts are available.

--- a/helm/hpcc/values.schema.json
+++ b/helm/hpcc/values.schema.json
@@ -146,6 +146,15 @@
       "description": "thor process",
       "type": "array",
       "items": { "$ref": "#/definitions/thor" }
+    },
+    "kibana": {
+      "description": "HPCC managed Elastic Stack deployment Kibana (log visualizer) component"
+    },
+    "elasticsearch": {
+      "description": "HPCC managed Elastic Stack deployment ElasticSearch (log processor) component"
+    },
+    "filebeat": {
+      "description": "HPCC managed Elastic Stack deployment Filebeat (log agent) component"
     }
   },
   "required": [

--- a/helm/hpcc/values.yaml
+++ b/helm/hpcc/values.yaml
@@ -8,7 +8,7 @@ global:
     #version: latest
     root: "hpccsystems"    # change this if you want to pull your images from somewhere other than DockerHub hpccsystems
     pullPolicy: IfNotPresent
-  
+
   # logging sets the default logging information for all components. Can be overridden locally
   logging:
     detail: 100
@@ -37,6 +37,68 @@ global:
   cost:
     moneyLocale: "en_US.UTF-8"
     perCpu: 0.126
+
+##Minimal instance of Elastic stack will be deployed alongside HPCC components
+##Elastic stack is comprised of several components which follow below and can be customized
+##by overwriting/setting helm chart values here
+
+##The elasticsearch component can be customized by modifying helm chart values here.
+elasticsearch:
+  enabled: true
+  description: "HPCC Managed Elasticsearch"
+  ##See https://github.com/elastic/helm-charts/blob/master/elasticsearch/values.yaml for all available options
+  antiAffinity: "soft"  #default is HARD, for minimal systems soft might be necessary
+  replicas: 1           #default is 3, for minimal systems 1 replicas should be adequate
+  minimumMasterNodes: 1 #default is 2, for minimal systems 1 master node should be adequate
+  labels: {"managedby" : "HPCC"}
+  #volumeClaimTemplate:
+    #accessModes: [ "ReadWriteOnce" ]
+    #resources:
+    #requests:
+    #  storage: 30Gi
+  #persistence:
+    #enabled: true
+    #labels:
+      # Add default labels for the volumeClaimTemplate fo the StatefulSet
+      #enabled: false
+    #annotations: {}
+
+##The filebeat component can be customized by modifying helm chart values here.
+filebeat:
+  description: "HPCC Managed filebeat"
+  ##See https://github.com/elastic/helm-charts/blob/master/filebeat/values.yaml for all available options
+  labels: {"managedby" : "HPCC"}
+  ## Allows you to add any config files in /usr/share/filebeat
+  ## such as filebeat.yml
+  #filebeatConfig:
+    #filebeat.yml: |
+      #filebeat.inputs:
+      #- type: container
+      #  paths:
+      #    - /var/log/containers/*.log
+        #processors:
+        #- add_kubernetes_metadata:
+            #host: ${NODE_NAME}
+            #matchers:
+            #- logs_path:
+                #logs_path: "/var/log/containers/"
+      #output.elasticsearch:
+      #  host: '${NODE_NAME}'
+      #  hosts: '${ELASTICSEARCH_HOSTS:elasticsearch-master:9200}'
+      #  index: "hpcc-%{[fields.log_type]}-%{[agent.version]}-%{+yyyy.MM.dd}"
+
+##The kibana component can be customized by modifying helm chart values here.
+kibana:
+  enabled: false
+  description: "HPCC Managed Kibana"
+  ##See https://github.com/elastic/helm-charts/blob/master/kibana/values.yaml for all available options
+  labels: {"managedby" : "HPCC"}
+  ## Allows you to add any config files in /usr/share/kibana/config/
+  ## such as kibana.yml
+  #kibanaConfig: {}
+  #   kibana.yml: |
+  #     key:
+  #       nestedkey: value
 
 ## storage:
 ##

